wasm/wasm_source/src/lib.rs

```diff
@@ -27,6 +27,170 @@ pub mod tx_init_account {
     }
 }
 
+/// A tx for a PoS bond that stakes tokens via a self-bond or delegation.
+#[cfg(feature = "tx_bond")]
+pub mod tx_bond {
+    use anoma_vm_env::tx_prelude::proof_of_stake::{bond_tokens, BondId};
+    use anoma_vm_env::tx_prelude::*;
+
+    #[transaction]
+    fn apply_tx(tx_data: Vec<u8>) {
+        let signed =
+            key::ed25519::SignedTxData::try_from_slice(&tx_data[..]).unwrap();
+        let bond =
+            pos::Bond::try_from_slice(&signed.data.unwrap()[..]).unwrap();
+
+        // TODO temporary for logging:
```

Comment:
all the logging in here is going to be removed/replaced with https://github.com/anoma/anoma/issues/462

Reply:
same for `tx_unbond` and `tx_withdraw` below